shogun-lib/src/main/java/de/terrestris/shogun/lib/service/FileService.java

```diff
@@ -51,6 +65,63 @@ public File create(MultipartFile uploadFile) throws Exception {
         return savedFile;
     }
 
+    /**
+     * Creates a new File entity in the Database.
+     * The file is stored as file on the .
+     *
+     * @param uploadFile
+     * @param writeToSystem
+     * @return
+     * @throws Exception
+     */
+    public File create(MultipartFile uploadFile, Boolean writeToSystem) throws Exception {
+        if (!writeToSystem) {
+            return this.create(uploadFile);
+        }
+
+        String uploadBasePath = uploadProperties.getPath();
+        String filename = uploadFile.getOriginalFilename();
+
+        FileUtil.validateFile(uploadFile);
+        File file = new File();
+        file.setFileType(uploadFile.getContentType());
+        file.setFileName(filename);
+        file.setActive(true);
+
+        File savedFile = this.create(file);
+
+        UUID fileUuid = savedFile.getFileUuid();
+
+        // Setup path and directory
+        String path = fileUuid + "/" + filename;
+        java.io.File fileDirectory = new java.io.File(uploadBasePath + "/" + fileUuid);
+        fileDirectory.mkdirs();
+
+        // Write multipart file data to target directory
+        byte[] fileByteArray = FileUtil.fileToByteArray(uploadFile);
+        java.io.File outFile = new java.io.File(fileDirectory, filename);
+        InputStream in = new ByteArrayInputStream(fileByteArray);
+
+        try (OutputStream out = new FileOutputStream(outFile)) {
+            IOUtils.copy(in, out);
+            LOG.info("Saved file with id {} to {}: ", savedFile.getId(), savedFile.getPath());
+        } catch (Exception e) {
+            LOG.error("Error when saving file {} to disk: " + e.getMessage(), savedFile.getId());
+            throw e;
```

Comment:
Do we have to manually rollback the creation here?

Reply:
either that, or we have to create and save the entity after the file was saved to disk

but in that case we could be left with a file without an entity in case `this.create()` fails...

Reply:
As we use the  uuid for the file location i don't think we can't upload the file and then create it. But the rollback should be a oneline...